src/cljdoc/render/index_pages.clj

```diff
@@ -99,17 +98,27 @@
   :args (spec/cat :versions (spec/coll-of :cljdoc.spec/version-entity)))
 
 (defn full-index
-  [versions]
+  [versions-tree]
   (->> [:div
         (layout/top-bar-generic)
         [:div.pa4-ns.pa2
          [:div#js--cljdoc-navigator]
          [:h1.mt5 "All documented artifacts on cljdoc:"]
-         (for [[group-id versions-for-group] (sort-by key (group-by :group-id versions))]
+         (for [[group-id groups-artifact-id->versions] versions-tree]
            [:div.cf
-            [:h2 group-id [:span.gray.fw3.ml3.f5 "Group ID"] ]
+            [:h2 group-id [:span.gray.fw3.ml3.f5 "Group ID"]]
             [:div.nl2.nr2
-             (for [[_a-id versions-for-artifact] (group-by :artifact-id versions-for-group)
-                   :let [latest (first (sort-by-version versions-for-artifact))]]
+             (for [[_a-id versions-for-artifact] groups-artifact-id->versions
+                   :let [latest (first versions-for-artifact)]]
                (artifact-grid-cell latest))]])]]
        (layout/page {:title "all documented artifacts — cljdoc"})))
+
+(defn versions-tree
```

Comment:
It doesn't seem optimal to have the `versions-tree` here in index-pages but is there a better place for it somewhere?

Reply:
Not too concerned about this. Maybe it could be moved into `cljdoc.util` but I don't really mind it being here. One thing that might be nice though: move the `sort-by-version` helper closer to this function since it's the only one using it.